crates/http/src/http.rs

```diff
@@ -60,33 +62,26 @@ pub trait HttpClient: Send + Sync {
         }
     }
 
-    fn proxy(&self) -> Option<&Uri>;
+    fn proxy(&self) -> Proxy;
 }
 
 /// An [`HttpClient`] that may have a proxy.
-#[derive(Deref)]
 pub struct HttpClientWithProxy {
-    #[deref]
     client: Arc<dyn HttpClient>,
-    proxy: Option<Uri>,
+    proxy: Proxy,
 }
 
 impl HttpClientWithProxy {
     /// Returns a new [`HttpClientWithProxy`] with the given proxy URL.
-    pub fn new(proxy_url: Option<String>) -> Self {
-        let proxy_url = proxy_url
-            .and_then(|input| {
-                input
-                    .parse::<Uri>()
-                    .inspect_err(|e| log::error!("Error parsing proxy settings: {}", e))
-                    .ok()
-            })
-            .or_else(read_proxy_from_env);
-
-        Self {
-            client: client(proxy_url.clone()),
-            proxy: proxy_url,
-        }
+    pub fn new(proxy: Proxy) -> Self {
+        let client = Arc::new(
+            isahc::HttpClient::builder()
+                .connect_timeout(Duration::from_secs(5))
+                .low_speed_timeout(100, Duration::from_secs(5))
+                .build()
+                .unwrap(),
+        );
+        Self { client, proxy }
```

Comment:
I've removed the client_builder and kept the client. Also, I've removed the default proxy setting from isahc client, but placed into each request.

Because when say, proxy setting is set to null, and each request has no proxy setting, the client default proxy setting will be used:

When request proxy setting is None, isahc client default proxy setting is used:
https://github.com/sagebind/isahc/blob/9d1edd475231ad5cfd5842d939db1382dc3a88f5/src/client.rs#L1016-L1037
https://github.com/sagebind/isahc/blob/9d1edd475231ad5cfd5842d939db1382dc3a88f5/src/config/request.rs#L51-L59

So, we will have to keep no default proxy settings is isahc client, to make requests respect the latest proxy setting in `Proxy`.